crates/bevy_math/src/primitives/dim3.rs

```diff
@@ -171,3 +206,76 @@ pub struct Capsule {
 }
 impl super::Primitive2d for Capsule {}
 impl Primitive3d for Capsule {}
+
+impl Capsule {
+    /// Create a new `Capsule` from a radius and half-length
+    pub fn new(radius: f32, half_length: f32) -> Self {
+        Self {
+            radius,
+            half_length,
+        }
+    }
+}
+
+/// A cone primitive.
+#[derive(Clone, Copy, Debug)]
+pub struct Cone {
+    /// radius of the base
+    pub radius: f32,
+    /// height of the cone
+    pub height: f32,
+}
+impl Primitive3d for Cone {}
+
+impl Cone {
+    /// Create a new `Cone` from a radius and height
+    pub fn new(radius: f32, height: f32) -> Self {
+        Self { radius, height }
+    }
+}
+
+/// A conical frustum primitive.
+/// A conical frustum can be created
+/// by slicing off a section of a cone.
+#[derive(Clone, Copy, Debug)]
+pub struct ConicalFrustum {
```

Comment:
This is also nice to have, but if we also want primitives to be used for rendering, a viewing frustum would be more useful. It would be defined by an aspect-ratio, fov and the near and far planes. But maybe that's a bit too rendering-specific for primitives? More opinions on this would be useful.

If we want to support different types of frusta, then perhaps a polygonal frustum with a regular polygon as a base could also be useful?

Reply:
@superdump also briefly touched on this here: https://github.com/bevyengine/bevy/pull/10466#issuecomment-1803406318

Reply:
Yep, we should split rendering frustra into a different PR.

Reply:
Rendering frustum most likely belongs in the bounding PR. Might be good to create some helper later to create a primitive shape you can render from the 6 plane representation (which is not exactly most convinient way to render them)